mipengine/node/monetdb_interface/common_actions.py

```diff
@@ -358,16 +359,14 @@ def drop_db_artifacts_by_context_id(context_id: str):
         The id of the experiment
     """
 
-    _drop_udfs_by_context_id(context_id)
-    # Order of the table types matter not to have dependencies when dropping the tables
-    table_type_drop_order = (
-        TableType.MERGE,
-        TableType.REMOTE,
-        TableType.VIEW,
-        TableType.NORMAL,
+    function_names = _get_function_names_query_by_context_id(context_id)
+
+    udfs_deletion_query = _get_drop_udfs_query(function_names) if function_names else ""
```

Comment:
Handle the if condition inside the method?

Same for the `_get_drop_tables_query`.